src/matlab2tikz.m

```diff
@@ -4005,17 +4010,37 @@ function warnFacetedInterp(m2t)
     else
         arrowLength = '{sqrt((\thisrow{u})^2+(\thisrow{v})^2)}';
     end
+
     plotOpts = opts_add(plotOpts, 'point meta', arrowLength);
+    plotOpts = opts_add(plotOpts, 'point meta min', '0');
 
     if showArrowHead
         %TODO: scale the arrows more rigorously to match MATLAB behavior
-        %There is a "MaxHeadSize" property (at least in R2014b) that plays a
-        %role in determining the quiver size.
+        %There is a "MaxHeadSize" property that plays a role.
+        % Currently, this is quite hard to do, since the size of the arrows
+        % is defined in pgfplots in absolute units (here we specify that those
+        % should be scaled up/down according to the data), while the data itself
+        % is in axis coordinates (or some scaled variant). I.e. we need the
+        % physical dimensions of the axis to compute the correct scaling!
+        userInfo(m2t, ['Please change the "arrowHeadSizeFactor" option', ...
+                       ' if the size of the arrows is incorrect.']);
+        arrowHeadSizeFactor = sprintf(m2t.ff, abs(m2t.cmdOpts.Results.arrowHeadSizeFactor));
+
+        % NOTE: `set(h, 'MaxHeadSize')` is bugged in HG1 (not in HG2 or Octave)
+        % according to http://www.mathworks.com/matlabcentral/answers/96754
+
+        % MaxHeadSize is said to be relative to the length of the quiver in the
+        % MATLAB documentation. However, in practice, there seems to be a SQRT
+        % involved somewhere (e.g. if u.^2 + v.^2 == 2, all MHS values >
+        % 1/sqrt(2) are capped to 1/sqrt(2)).
```

Comment:
The `sqrt` is certainly related to Pythagoras (2-norm, Euclidean distance, length, ... of the arrow).

However, what the comment meant to show is that, the `MaxHeadSize`(MHS) is not relative to the actual length of an arrow, but rather that if you have an arrow with length 2, that you see two behaviors:
 * `MaxHeadSize < 1/sqrt(2)`: changing MHS changes the size of the arrow, and
 * `MaxHeadSize >= 1/sqrt(2)`: changing MHS has no effect on the figure.

It's the easiest to see if you make yourself a little quiver plot and try it out in MATLAB (>= 2014b, due to the bugs in <=2014a) by interactively changing the MHS.

I.e. `MHS > 1/sqrt(2)` is capped at `1/sqrt(2)` it seems, but I didn't investigate further how all of that fits together to replicate the same behavior in TikZ code, but instead I wrote it down in that comment such that I (or somebody else) can pick this up later.